src/Pulsar.Client/Schema/AvroSchema.fs

```diff
@@ -35,7 +49,16 @@ type internal AvroSchema<'T> private (stringSchema) =
         use stream = new MemoryStream(bytes)
         avroReader.Read(Unchecked.defaultof<'T>, BinaryDecoder(stream))        
     override this.GetSpecificSchema stringSchema =
-        AvroSchema(stringSchema) :> ISchema<'T>
+        let writtenSchema = Schema.Parse(stringSchema)
+        if avroReader :? SpecificDatumReader<'T> then
+            AvroSchema(schema, SpecificDatumReader(writtenSchema, schema), avroWriter) :> ISchema<'T>
+        else
+            //Avro doesnt figure that the written classname might be different from the reader classname
+            //Seems like it might be a bug in ReflectReader, but this works around that
```

Comment:
Not sure if found an avro bug here or not.  They do talk about manually registering in the docs, but not in this context: https://github.com/apache/avro/tree/master/lang/csharp/src/apache/main/Reflect#manual-registration

Reply:
I see, in that case you can try raising an issue in their Jira tracker if this is an expected behaviour. As for us it looks this workaround is needed anyway, so merging the PR, big thanks!

Reply:
fyi created [AVRO-2873](https://issues.apache.org/jira/browse/AVRO-2873) in the Apache JIRA,